### ISSUE TITLE
Add columns to randoms required to run through fiberassign

### ISSUE THREAD

**Issue by ashleyjross** [desihub/desitarget#597]
AJR added the following to prepare random files for fiberassign (in https://github.com/desihub/LSS/blob/master/Sandbox/miniSVcat/cattools.py)

	rmtl['TARGETID'] = np.arange(len(rall))
	rmtl['DESI_TARGET'] = np.ones(len(rall),dtype=int)*2
	rmtl['SV1_DESI_TARGET'] = np.ones(len(rall),dtype=int)*2
	rmtl['NUMOBS_INIT'] = np.zeros(len(rall),dtype=int) #these numobs aren't quite correct
	rmtl['NUMOBS_MORE'] = np.ones(len(rall),dtype=int)
	rmtl['PRIORITY'] = np.ones(len(rall),dtype=int)*3400
	rmtl['OBSCONDITIONS'] = np.ones(len(rall),dtype=int) #this needs to be made to work for all tile types
	rmtl['SUBPRIORITY'] = np.random.random(len(rall))

**Comment by geordie666** [comment]
Thanks @ashleyjross. Can you clarify:

Do you actually want values _exactly_ like ```rmtl['PRIORITY'] = np.ones(len(rall),dtype=int)*3400``` or do you want more sophisticated behavior, such as a random being assigned an actual, realistic target type (such as "QSO", "ELG", overlapping "QSO/ELG" etc.) and the associated `PRIORITY` and `OBSCONDITIONS` for that target type?

Perhaps a reasonable answer is "let's do the simple thing in my comment for now, and work towards something more sophisticated later"?

**Comment by ashleyjross** [comment]
For these, my idea is that we will purely be probing geometry, so all points get treated exactly the same (and it is important, e.g., to set things so that this is the case no matter the tile type). Above is just what I put together to make things work in my first attempt. The important information is what gets written to the FAVAIL HDU (not the actual assignment).

**Comment by geordie666** [comment]
@ashleyjross: I now add the following columns automatically to the randoms:
```
TARGETID
DESI_TARGET
NUMOBS_INIT
NUMOBS_MORE
PRIORITY
OBSCONDITIONS
SUBPRIORITY
```
So, for DR9 (and henceforth), those columns will exist automatically in the random catalogs.

Hopefully these will be useful, because they'll have realistic values that are assigned in the same way they're assigned for the targets (for example, `TARGETID` will be correctly assembled from the brick information).

**Comment by geordie666** [comment]
I'll close this soon.

**Comment by djschlegel** [comment]
For those fields other than TARGETID, what would they mean, or would they be empty?

**Comment by geordie666** [comment]
Most of the fields are set as requested by Ashley, above, unless the value of one field is dependent on another field (i.e. values like ``PRIORITY`` and ``NUMOBS_INIT`` depend on the target class). Broadly:

- `DESI_TARGET` is set appropriately to the highest priority target (i.e. LyA quasars).
- `PRIORITY` and `NUMOBS_` are set as if for the highest priority target (i.e. LyA quasars).
- `OBSCONDITIONS` is set to "all observing conditions are allowed".
-  `SUBPRIORITY` is set using the same seed as for the targets.

It sounds like we may be moving towards running things through `select_mock_targets` instead of using the randoms more directly, but it's easy enough for me to comment out my new functionality if the random files become cumbersome.

**Comment by ashleyjross** [comment]
Thanks for these additions!

 

I don't understand this comment:

 

" It sounds like we may be moving towards running things through select_mock_targets instead of using the randoms more directly"

 

I'm definitely planning on using these randoms directly.

 

Cheers,

Ashley

**Comment by geordie666** [comment]
Addressed in #606.
